.github/workflows/continuous-integration.yml

```diff
@@ -1,6 +1,18 @@
 name: Continuous Integration
 
-on: [push, pull_request]
+on:
+  pull_request:
+    paths:
+      - 'src/**'
+      - 'tests/**'
+      - composer.json
+  push:
+    branches:
```

Comment:
Is there a reason why you included the `branches` section for the `push` action?

Reply:
If I'm not mistaken, because it's a while back, this allows for CI to run on pushes to all branches, not only the master branch. I can remove it if it's preferred.

Reply:
[According to the documentation](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs), if you specify the branches configuration, you are telling GitHub to execute the pipeline only when the listed branches are affected.

Reply:
For this reason, I suppose it's better to avoid specifying the branches.